platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerContractTest.java

```diff
@@ -0,0 +1,147 @@
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.codice.ddf.ui.admin.api.util.PropertiesFileReader;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * TODO: DDF-2731
+ * Temporary test class to be removed and refactored as part of moving {@link PropertiesFileReader}
+ * to a common module. The unit tests will need to be reworked and sorted appropriately.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class GuestClaimsHandlerContractTest {
+    private static final String IMMUTABLE_CLAIM_1 = "immutableClaim1";
+
+    private static final String IMMUTABLE_CLAIM_2 = "immutableClaim2";
+
+    private static final String GUEST = "guest";
+
+    private static final String PROFILES_DIRECTORY = "profiles";
+
+    private static final String PROFILES_BANNER_DIRECTORY = "profileBanners";
+
+    private static final String AVAILABLE_CLAIMS_FILE = "availableClaims";
+
+    private static final String PROFILE_A = "profileA";
+
+    private static final String PROFILE_A_VALUE = "profileAValue";
+
+    private static final String PROFILE_B = "profileB";
+
+    private static final String PROFILE_B_VALUE = "profileBValue";
+
+    @Mock
+    private PropertiesFileReader mockPropertiesFileReader;
+
+    private GuestClaimsHandlerExt handlerExt;
+
+    @Before
+    public void setup() throws Exception {
+        when(mockPropertiesFileReader.loadSinglePropertiesFile(eq(AVAILABLE_CLAIMS_FILE))).thenReturn(
+                ImmutableMap.of(IMMUTABLE_CLAIM_1, GUEST, IMMUTABLE_CLAIM_2, GUEST));
+
+        when(mockPropertiesFileReader.loadPropertiesFilesInDirectory(eq(PROFILES_DIRECTORY))).thenReturn(
+                Arrays.asList(ImmutableMap.of(PROFILE_NAME_KEY,
+                        PROFILE_A,
+                        PROFILE_A,
+                        PROFILE_A_VALUE),
+                        ImmutableMap.of(PROFILE_NAME_KEY, PROFILE_B, PROFILE_B, PROFILE_B_VALUE),
+                        ImmutableMap.of(PROFILE_B, PROFILE_B_VALUE)));
+
+        when(mockPropertiesFileReader.loadPropertiesFilesInDirectory(eq(PROFILES_BANNER_DIRECTORY))).thenReturn(
+                Arrays.asList(ImmutableMap.of(PROFILE_NAME_KEY,
+                        PROFILE_A,
+                        PROFILE_A,
+                        PROFILE_A_VALUE),
+                        ImmutableMap.of(PROFILE_NAME_KEY, PROFILE_B, PROFILE_B, PROFILE_B_VALUE)));
+
+        handlerExt = new GuestClaimsHandlerExt(mockPropertiesFileReader,
+                Arrays.asList(IMMUTABLE_CLAIM_1, IMMUTABLE_CLAIM_2),
+                PROFILES_DIRECTORY,
+                PROFILES_BANNER_DIRECTORY,
+                AVAILABLE_CLAIMS_FILE);
+
+        handlerExt.init();
+    }
+
+    @Test
+    public void testFieldNullability() throws Exception {
```

Comment:
Should we have a test similar to this that tests when the `selectedClaimsProfileName` is the default?